include/cute/algorithm/reorder.hpp

```diff
@@ -165,21 +165,41 @@ reorder_impl(ReorderAtom               const& atom,
   constexpr int values = size(SLayout{}) / size<0>(SLayout{});
   constexpr int vchunk = sizeof_bits_v<RegistersSrc> / sizeof_bits_v<SType>;
 
-  // Calculate mapping from src val -> dst val on a chunk-by-chunk basis. Unlike a plain copy, there is no intrinsic
-  //   correspondence of src/dst values for subgroup reorders.
-  auto rlayout = coalesce(composition(right_inverse(dlayout), slayout));                 // src index -> dst index
-  auto vrlayout = composition(composition(Layout<Shape<_SG, Int<values>>, Stride<_0, _1>>{},
-                                          rlayout),
-                              Layout<Shape<_1, Int<values>>, Stride<_0, _SG>>{});        // src val -> dst val
-
-  CUTE_UNROLL
-  for (int sv = 0; sv < values; sv += vchunk) {
-    auto pS = recast_ptr<RegTypeSrc>(src.data() + sv);
-    auto pD = recast_ptr<RegTypeDst>(dst.data() + vrlayout(sv));
-
-    detail::explode(detail::CallReorder<ReorderAtom>{},
-                    pS, make_int_sequence<RegNumSrc>{},
-                    pD, make_int_sequence<RegNumDst>{});
+  static constexpr bool has_broadcast = (size(DLayoutWI{}) > size(SLayoutWI{}));
```

Comment:
Sorry, I'm confused about how we should load elements in the source tensor before broadcasting them.
Can we read contiguous values for coalesced access, and then would reorder take care of the reshuffle based on the `DLayout`?

For example, if we read 64 contiguous values (4 per work item, such that WI 0 gets elements 0, 16, 32, 48), can this implementation do an arbitrary reshuffle, such as the one below?

WI 0 & WI 1 have 0, 8, 16, 24, 32, 40, 48, 56
WI 2 & WI 3 have 1, 9, 17, 25, 33, 49, 57
and so forth...


Thanks!